src/content/Kentico.Kontent.Boilerplate/Startup.cs

```diff
@@ -35,27 +33,29 @@ public void ConfigureServices(IServiceCollection services)
 
             // Register the IConfiguration instance which ProjectOptions binds against.
             services.Configure<ProjectOptions>(Configuration);
-
+            
             var deliveryOptions = new DeliveryOptions();
             Configuration.GetSection(nameof(DeliveryOptions)).Bind(deliveryOptions);
 
-            services.AddSingleton<IWebhookListener>(sp => new WebhookListener());
-            services.AddSingleton<IDependentTypesResolver>(sp => new DependentFormatResolver());
-            services.AddSingleton<ICacheManager>(sp => new ReactiveCacheManager(
-                sp.GetRequiredService<IOptions<ProjectOptions>>(),
-                sp.GetRequiredService<IMemoryCache>(),
-                sp.GetRequiredService<IDependentTypesResolver>(),
-                sp.GetRequiredService<IWebhookListener>()));
-            services.AddScoped<KenticoKontentSignatureActionFilter>();
-
-            services.AddSingleton<IDeliveryClient>(sp => new CachedDeliveryClient(
-                sp.GetRequiredService<IOptions<ProjectOptions>>(),
-                sp.GetRequiredService<ICacheManager>(),
-                DeliveryClientBuilder.WithOptions(_ => deliveryOptions)
+            services.AddScoped<SignatureActionFilter>();
+
+            IDeliveryClient BuildBaseClient(IServiceProvider sp) => DeliveryClientBuilder
+                .WithOptions(_ => deliveryOptions)
                 .WithTypeProvider(new CustomTypeProvider())
                 .WithContentLinkUrlResolver(new CustomContentLinkUrlResolver())
-                .Build())
-               );
+                .Build();
+
+            // Use cached client version based on the use case
```

Comment:
The caching client should be default. After all, you need to configure create a webhook before you can use cache with sliding expiration.